Use autogenerated bindings for egl and glx

ecoal95 opened the pull request:
This will remove the rust-glx dependency (which does just this), and the rust-egl dependency, which uses manual bindings.

I wonder if just updating rust-egl to generate the bindings is a better alternative, since azure, the gonk port, and glutin_app use it.

What do you think @glennw?

glennw commented:
@ecoal95 I like your suggestion of updating rust-egl to use the auto bindings, that sounds like a great idea to me.

ecoal95 commented:
The problem I see is that we're going to have to make a lot of changes to existing code in order to land it, but I think it's worth it.

glennw commented:
Sorry I'm a bit out of the loop on this - what effect does this have on updating to use euclid? Are you on IRC - it might be easier to discuss there rather than in comments.

ecoal95 commented:
Nah, it should have no effect. In fact, I think https://github.com/servo/servo/pull/6423 is already mergeable.

What happened was the following:

I updated `rust-offscreen-rendering-context` to use autogenerated bindings recently, and in that pr I updated to the latest upstream.

Unfortunately I can't cross-compile it with the feature used in servo because of https://github.com/servo/servo/issues/6384.

The bindings generated by gl_generator and rust-egl seem to be conflictive (gl_generator defines `EGLDisplay` as `*const c_void`, while our egl bindings use `*mut`, so I can't interact with layers.

At this point I splitted the commit and as I said before it should be mergeable, but to solve that conflict (later) I should probably use a transmute or make egl autogenerate bindings.

That's totally another topic with euclid, it was just coincidence, but since I want to resolve the issue, and you told me the rust-egl bindings should be replaced, I think it's worth it to generate them, and since this week I will have time, I decided to do it.

ecoal95 commented:
Sorry for the confusion, by the way :/

glennw commented:
@ecoal95 Ah, I understand now :) I'll retry #6423 now.

ecoal95 commented:
Closing this for now
